### PR TITLE
verify that optprof scripts are not broken

### PR DESCRIPTION
**Pull request by jmarolf**

### CODE REVIEW

File: build/scripts/test-build-correctness.ps1 [hunk 1]
```diff
@@ -36,7 +36,31 @@ try {
...
+    # Verfiy the state of creating run settings for optprof
+    Write-Host "Checking run generation for optprof"
+
+    # set environment variables
+    if (-not (Test-Path env:SYSTEM_TEAMPROJECT)) { $env:SYSTEM_TEAMPROJECT = "DevDiv" }
+    if (-not (Test-Path env:BUILD_REPOSITORY_NAME)) { $env:BUILD_REPOSITORY_NAME = "dotnet/roslyn" }
+    if (-not (Test-Path env:BUILD_SOURCEBRANCHNAME)) { $env:BUILD_SOURCEBRANCHNAME = "test" }
+    if (-not (Test-Path env:BUILD_BUILDID)) { $env:BUILD_BUILDID = "42.42.42.42" }
+    if (-not (Test-Path env:BUILD_SOURCESDIRECTORY)) { $env:BUILD_SOURCESDIRECTORY = $RepoRoot }
+    if (-not (Test-Path env:BUILD_STAGINGDIRECTORY)) { $env:BUILD_STAGINGDIRECTORY = $configDir }
```
Comment (jaredpar) [inline build/scripts/test-build-correctness.ps1]: Why only conditionally setting these variables? Isn't it simpler to juts always set them?

File: build/scripts/test-build-correctness.ps1 [hunk 2]
```diff
@@ -36,7 +36,31 @@ try {
...
+
+    # create a fake BootstrapperInfo.json file
+    $bootstrapperInfoFolder = Join-Path $configDir "MicroBuild\Output"
+    if(!(Test-Path -Path $bootstrapperInfoFolder )){
+        New-Item -ItemType directory -Path $bootstrapperInfoFolder
+    }
```
Comment (jaredpar) [inline build/scripts/test-build-correctness.ps1]: Just use `Create-Directory` here vs the test and create path.

File: build/scripts/test-build-correctness.ps1 [hunk 3]
```diff
@@ -36,7 +36,31 @@ try {
...
+    $bootstrapperInfoPath = Join-Path $bootstrapperInfoFolder "BootstrapperInfo.json"
+    $bootstrapperInfoContent = "[{""VSBuildVersion"":  ""16.0.28309.54""}]"
```
Comment (jaredpar) [inline build/scripts/test-build-correctness.ps1]: Will hard coding the version here get us into trouble down the road?

File: build/scripts/test-build-correctness.ps1 (hunk 1 above)
Comment (jmarolf) [reply build/scripts/test-build-correctness.ps1]: They will be set in the ADO build so this is for running them on your local machine

File: build/scripts/test-build-correctness.ps1 (hunk 3 above)
Comment (jmarolf) [reply build/scripts/test-build-correctness.ps1]: I could use 42.42.42.42 here instead to better indicate that this is a non-real number.  In the signed build this number comes from Micobuild so if this breaks its not anyones fault who checked into this repo.

Comment (jmarolf) [reply build/scripts/test-build-correctness.ps1]: This number is just read and copied verbatim into the runsettings file

**Pull request merged**